os/fs/driver/mtd/smart.c

```diff
@@ -1341,6 +1341,41 @@ static int smart_setsectorsize(FAR struct smart_struct_s *dev, uint16_t size)
 	return -ENOMEM;
 }
 
+/****************************************************************************
+ * Name: smart_byte_to_block_write
+ *
+ * Description: To support write some bytes to flash if architecture
+ *              doesn't support bytewrite, first read entire of sector,
+ *              and then do blockwrite after merge rwbuffer & buffer.
+ *
+ ****************************************************************************/
+static int smart_byte_to_block_write(FAR struct smart_struct_s *dev, size_t offset, int nbytes, FAR const uint8_t *buffer)
+{
+	uint32_t startblock;
+	uint16_t nblocks;
+	int ret = OK;
+	/* First calculate the start block and number of blocks affected. */
+	
+	startblock = offset / dev->geo.blocksize;
```

Comment:
Is there any possibility that dev->geo.blocksize would be zero?

Reply:
yes, it will never happen.